scene/2d/parallax_layer.cpp

```diff
@@ -123,12 +123,12 @@ void ParallaxLayer::set_base_offset_and_scale(const Point2 &p_offset, real_t p_s
 	Point2 new_ofs = (screen_offset + (p_offset - screen_offset) * motion_scale) + motion_offset * p_scale + orig_offset * p_scale;
 
 	if (mirroring.x) {
-		double den = mirroring.x * p_scale;
+		real_t den = mirroring.x * p_scale;
 		new_ofs.x -= den * ceil(new_ofs.x / den);
 	}
 
 	if (mirroring.y) {
-		double den = mirroring.y * p_scale;
+		real_t den = mirroring.y * p_scale;
```

Comment:
Why is this necessary? Not trying to second-guess you, just wondering because I've never seen real_t used before.

Reply:
`real_t` is the abstract type that Godot uses for vectors and much more. It can be defined as either float or double. In the audio code you'll see `float` because you don't need more precision than that when working with audio.